### ISSUE TITLE
kodi-addon-checker.log (Found non whitelisted file ending in filename) reported on PR test

### ISSUE THREAD

**Issue by anxdpanic** [xbmc/addon-check#156]
Started integrating TravisCI/addon-check into some repositories, currently quite a few false problems/warnings. 
It looks like the ```.git``` folder is ignored in master(which would resolve most false reports), but I don't see anything addressing the log file in master.
See: https://travis-ci.org/jdf76/plugin.video.youtube/jobs/485145419#L520

**Comment by Razzeee** [comment]
Jep, needs to be ignored by the full name I guess.

Still fells like we need more then `--PR` and not setting that.
As far as I know we have at least three cases:
- Running against a repo
- Running as PR (one addon folder)
- Running it not as a PR against your folder

Might be nicer if we can change that commandline option.
And the things we need to change for the last case should only be removing git from the scan.

See https://github.com/xbmc/addon-check/issues/123 for the other problem.

**Comment by Rechi** [comment]
> `ERROR: plugin.video.twitch addon already exists with version: 2.4.1 in isengard branch`

That's exactly the use case of the `--PR` command line argument. It's not expected that you run your checks always with `--PR`. It's only for our repos to check that the version is actually increased in a PR.

**Comment by anxdpanic** [comment]
Okay, so ```--PR``` shouldn't be used on PR's outside of the official repo?
I'm using it on PR's like in the above log, based on the argument description and https://github.com/xbmc/addon-check/blob/master/.travis-addon.yml.example

**Comment by Rechi** [comment]
Yes you shouldn't use that. The only location it is used the following (forcing strictly higher version).
https://github.com/xbmc/addon-check/blob/09298318e9423940c0b740e3c5acc775b0d7077a/kodi_addon_checker/check_old_addon.py#L46-L59

**Comment by anxdpanic** [comment]
@Rechi Thank you for the clarification. I've updated where necessary 👍

**Comment by Rechi** [comment]
@anxdpanic a new version was released which includes the `.git` folder fix.

**Comment by anxdpanic** [comment]
I was looking at adding something to resolve the mentioned log/license warnings
I had a couple ideas on how to approach this;

1. ```--exclude-file``` and ```--exclude-file-ext``` to exclude file/folder and extensions from permissions and extension whitelist
2. ```--whitelist-file``` and ```--whitelist-file-ext``` to exclude from extension whitelist
3. both

This would also allow for .pyo/.pyc to be ignored as well when done against a folder

Any suggestions on how this should be approached?

**Comment by anxdpanic** [comment]
I was able to resolve the initial issue (log and license) within travis ci, 
https://github.com/jdf76/plugin.video.youtube/blob/master/.travis.yml#L8
https://travis-ci.org/jdf76/plugin.video.youtube/builds/491055230

**Comment by anxdpanic** [comment]
This issue was resolved locally and ci outside of kodi-addon-checker
